### DISCUSSION TITLE
get transactions history

### DISCUSSION THREAD

**Discussion by MU12351235** [Python-Cardano/pycardano#235] [Q&A]
Hello
Hope you feel good
I am working on a Cardano related project
Now I want to get an address deposits history
Please help me how to do it?

**Comment by MU12351235** [comment]
Please answer

**Comment by cffls** [comment]
PyCardano is not able to provide the transaction history for an address. However, there are other tools you can use. For example, Blockfrost provides an endpoint for getting all transactions related to an address: https://docs.blockfrost.io/#tag/Cardano-Addresses/paths/~1addresses~1%7Baddress%7D~1transactions/get

**Comment by MU12351235** [comment]
this is good
But in order to find out which of the transactions are deposits, I have to make API call for each one, and this is very time-consuming.
Is there a way to get all the deposits?
or getting transactions with a deposit or withdrawal feature?
Thank you for your response.

**Reply by cffls** [reply]
You can probably also use [cardano-db-sync](https://github.com/input-output-hk/cardano-db-sync) or [kupo](https://github.com/CardanoSolutions/kupo). I would ask this question on https://cardano.stackexchange.com/, since this question is actually out of the scope of pycardano.
